src/esp/physics/CollisionGroupHelper.h

```diff
@@ -0,0 +1,236 @@
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_PHYSICS_COLLISIONGROUPHELPER_H_
+#define ESP_PHYSICS_COLLISIONGROUPHELPER_H_
+
+#include <Corrade/Utility/Assert.h>
+#include <Magnum/Magnum.h>
+#include <map>
+#include <string>
+#include <vector>
+#include "esp/core/Check.h"
+#include "esp/core/esp.h"
+
+/** @file
+ * @brief Class @ref esp::physics::CollisionGroupHelper is a convenience class
+ * with all static functions providing an interface for customizing collision
+ * masking behavior for simulated objects. Enum Class @ref
+ * esp::physics::CollisionGroup defines available collision groups.
+ */
+
+namespace esp {
+namespace physics {
+
+/*
+  For reference, here are the standard Bullet groups. It's probably good to
+  stay somewhat consistent with them, in case users are already using them.
+        enum CollisionFilterGroups
+        {
+                DefaultFilter = 1,
+                StaticFilter = 2,
+                KinematicFilter = 4,
+                DebrisFilter = 8,
+                SensorTrigger = 16,
+                CharacterFilter = 32,
+                AllFilter = -1  //all bits sets: DefaultFilter | StaticFilter |
+  KinematicFilter | DebrisFilter | SensorTrigger
+        };
+*/
+
+/**
+ * @brief Defined available collision groups with bitwise integer correspondance
+ * for use in collision filtering masks.
+ *
+ * Contains several named default groups used by Habitat-sim and a set of unused
+ * user groups for custom behavior.
+ */
+enum class CollisionGroup {
+  //! Default group for unspecified object (e.g. raycast)
+  Default = 1,
+  //! STATIC objects should not change state
+  Static = 1 << 1,
+  //! KINEMATIC objects are updated manually
+  Kinematic = 1 << 2,
+  //! Free (i.e., DYNAMIC) objects are integrated by dynamic simulation
+  FreeObject = 1 << 3,
+  //! convenience groups to separate robots from free objects
+  Robot = 1 << 4,
+  //! convenience group for objects which should not collide with anything
+  Noncollidable = 1 << 5,
+
+  // User configurable groups with "CollisionGroup::Default" behavior unless
+  // modified
+  UserGroup1 = 1 << 6,
+  UserGroup2 = 1 << 7,
+  UserGroup3 = 1 << 8,
+  UserGroup4 = 1 << 9,
+  UserGroup5 = 1 << 10,
+  UserGroup6 = 1 << 11,
+  UserGroup7 = 1 << 12,
+  UserGroup8 = 1 << 13,
+  UserGroup9 = 1 << 14,
+
+  //! convenience group for objects which should collide with everything except
+  //! Noncollidable
+  AllFilter = -1
+};
+
+/**
+ * @brief A convenience class with all static functions providing an interface
+ * for customizing collision masking behavior for simulated objects.
+ */
+class CollisionGroupHelper {
+  //! maps custom names to collision groups
+  static std::map<std::string, CollisionGroup> collisionGroupNames;
+  //! maps collision groups to collision group filter masks
+  static std::map<CollisionGroup, int> collisionGroupMasks;
```

Comment:
Collision groups are effectively an unsigned in. Doesn't that mean this should really just be an array?